### PR TITLE
E2E for Cilium CNI without Kubeproxy

### PR DESCRIPTION
**Pull request by adiantum**
For tech preview we are planning to using cilium without kube-proxy.
To get some validation these changes add E2E tests for Cilium CNI without kube-proxy to make it able to validate the configuration.
Changes:
 - add manifest scripts for Cilium CNI with kube-proxy replacement
 - add CAPX manifests with skip phase kube-proxy install
 - change makefile: add target for Cilium CNI with kube-proxy replacement E2E tests

Tests:
E2E tested on selected K8S versions. All tests pass except HostPort related tests which is well-known issue for Cilium CNI. 

```release-note
NONE
```

### CODE REVIEW

File: templates/cluster-template-no-kubeproxy-ccm.yaml [hunk 1]
```diff
@@ -0,0 +1,551 @@
+apiVersion: v1
```
Comment (yannickstruyf3) [inline templates/cluster-template-no-kubeproxy-ccm.yaml:1]: Do we need specific templates for this configuration?

**Review by yannickstruyf3 — CHANGES_REQUESTED** [review]

**Review by tuxtof — CHANGES_REQUESTED** [review]
templates/* need to be removed

**Comment by adiantum** [comment]
> templates/* need to be removed

Fixed

**Review by yannickstruyf3 — APPROVED** [review]

**Review by tuxtof — APPROVED** [review]
LGTM

File: test/e2e/data/cni/cilium/cilium-no-kubeproxy.yaml [hunk 2]
```diff
@@ -0,0 +1,1056 @@
+---
```
Comment (deepakm-ntnx) [inline test/e2e/data/cni/cilium/cilium-no-kubeproxy.yaml:1]: pls document the location from where we got this yaml

Comment (adiantum) [reply test/e2e/data/cni/cilium/cilium-no-kubeproxy.yaml:1]: @deepakm-ntnx It's documented inline: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/310/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R57

**Pull request merged**